frontend/src/pages/modelServing/customServingRuntimes/utils.ts

```diff
@@ -89,7 +89,20 @@ export const getServingRuntimeFromTemplate = (
   } catch (e) {
     return undefined;
   }
-  return template.objects[0];
+
+  // Add apiProtocol annotation if exists in template
+  const apiProtocolAttribute = 'opendatahub.io/apiProtocol';
+  const servingRuntimeObj = { ...template.objects[0] };
+  const metadata = { ...template.objects[0].metadata };
+
+  if (metadata.annotations && template.metadata.annotations?.[apiProtocolAttribute]) {
```

Comment:
Great, here you are error handling and won't select any value that is not one of the desired ones, this is the way to go.